cps/CP43.md

```diff
@@ -0,0 +1,28 @@
+---
+cp: 43
+title: Add CFG and DOT to the Asset Registry
+authors: Nuno
+contributors: 
+proposal-type: CP3
+date-proposed: 2023-04-13
+date-ended: 
+status: voting
+---
+
+## Content
+* Register CFG and DOT in the Asset Registry
+
+When we first added XCM to our runtimes (first Altair and later Centrifuge), the first token we supported was Acala's ```AUSD``` and for that we just added ```AUSD``` as a hardcoded variant of our ```CurrencyId``` enum type. That type enumerates all the currency ids that the Centrifuge chain knows about or is able to handle. Later, improving our XCM setup, we introduced the ORML asset registry for two purposes: 
+
+1) So that we could support new tokens through XCM by simply registering them in the asset registry instead of hardcoding them in the CurrencyId type and thus not require runtime upgrades to support new tokens.
+2) So that we had a standardised way of querying metadata for all tokens that Centrifuge handles, i.e, both native tokens like CFG and Tranche tokens, as well as "foreign" tokens like AUSD, USDC, etc.
+ 
+This motion is now a step into cleaning up that first legacy, "hardcoding"-based approach; we do that by dropping those hardcoded currency variants (```AUSD``` and ```KSM``` (the latter only applicable on Altair)) and have them registered as ```Currency::ForeignAsset(<asset_id>)``` in the asset registry instead. The next step will be to migrate all balances that are now under ```CurrencyId::AUSD```  and ```CurrencyId::KSM``` to ```CurrencyId::ForeignAsset(<aUSD_asset_id>)``` and ```CurrencyId::ForeignAsset(<ksm_asset_id>)```, respectively.
+
+Proposal hash: ```0x6de31686f6a4fbf1584f91e45961862f44e5a692b23a122bc29e26c5e2275d60```
+
+GitHub issue covering this work: https://github.com/centrifuge/centrifuge-chain/issues/826
+
+## Governance process for this proposal
```

Comment:
Should we add also the step:
-  Create a Proposal post forum?

Reply:
I don't think it's relevant to mention the Proposal Post in the Governance process for this CP - we haven't done it in the previous CPs (RUs). See [here](https://github.com/centrifuge/cps/blob/main/cps/CP26.md).